stdlib/public/core/Collection.swift

```diff
@@ -398,6 +392,15 @@ extension Collection {
   }
 }
 
+/// Supply a default `next()` method for `Collection` models that
+/// use some model of `Strideable` as their `Index`.
+extension Collection where Index : Strideable {
+  @warn_unused_result
+  public func next(i: Index) -> Index {
+    return i.advanced(by: 1)
```

Comment:
I guess at some point should we have the default _failEarlyRangeCheck use Comparable to do bounds checks? The default currently does nothing because of legacy reasons.